recipes/Code_Gen_from_Tests/Code_Gen_from_Tests.ipynb

```diff
@@ -0,0 +1,496 @@
+    "\n",
+    "replicate_response = granite_via_replicate.invoke(prompt)\n",
+    "\n",
+    "print(f\"Granite response from Replicate: {replicate_response}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can find a good implementation of `Rational` [`rational-good-example.py`, in the GitHub repo](https://github.com/ibm-granite-community/granite-code-cookbook/blob/main/recipes/Code_Gen_from_Tests/rational-good-example.py).\n",
+    "\n",
+    "How closely does the Granite output match this code? Does it appear to satisfy the requirements for rationals in the [Wikipedia article](https://en.wikipedia.org/wiki/Rational_number)? \n",
+    "\n",
+    "If important logic is missing, try changing the `prompt`, which is currently very generic:\n",
+    "1. Make the prompt more specific about the properties of rational numbers.\n",
+    "2. Add the link to the Wikipedia page mentioned above (which is also mentioned in the test code comments).\n",
+    "\n",
+    "Try any changes to the prompt one at a time to see their relative impact. Also, it's useful to run the query several times for each prompt change to see how the results vary."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Do the Tests Pass??\n",
+    "\n",
+    "Let's run the tests! Execute the next cell to create a `rational` directory and files we need:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!rm -rf rational\n",
+    "!mkdir -p rational\n",
+    "!echo \"from .rational import Rational\" > rational/__init__.py\n",
+    "!ls -al rational"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that the test code expects to find a `Rational` type in the `rational` package, which is why we did some of the steps in the previous cell.\n",
+    "\n",
+    "Now copy the generated `Rational` class definition in the output above and paste it in between the `\"\"\"` quotes in the next cell. _If the generated code has `\"\"\"` comments_, either delete them or add a `\\` in front of the first of the three `\"` _in each case_, like we did for the tests above.\n",
+    "\n",
+    "**Don't** include the test code, any markdown, or other text that was part of the generated output!\n",
+    "\n",
+    "**Do** include appropriate `import` statements, e.g., you may see that the generated code calls `gcd` (_greatest common divisor_), which would require this import `from math import gcd`. (If you aren't sure why `gcd` is useful, see the Wikipedia article linked above about the properties of rational numbers.)\n",
+    "\n",
+    "Is the indentation correct in the code? Make sure each level is properly indented."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rational = \"\"\"\n",
+    "\n",
+    "\"\"\""
```

Comment:
```suggestion
    "rational = '''\n",
    "\n",
    "'''"
```

Use triple single quotes to make it easy for the user to paste in the generated class.

Reply:
Same as above. _However_, I did use the hack `\"""`, which I replaced with your suggestion of `'''`.